### DISCUSSION TITLE
The problem of mapping odf values in Rodriguez space

### DISCUSSION THREAD

**Discussion by ZXJ2024** [neperfepx/neper#943] [Q&A] [answered]
Hello,
I am using the following command to assign an ODF value to grid cells in Rodrigues space:

neper -T -n 1 -domain "rodrigues(cubic)" -o fr
neper -M fr.tess -cl 0.05
neper -T -n 1000 -ori "odf(mesh=file(fr.msh),val=file(odf.txt))" -crysym cubic
However, I am unclear about how the odf.txt file in the tutorial is created. The file appears to be a 1D array, but the experimental data I obtained is in the following format:

% phi1  Phi  phi2  value

0.00000  0.00000  0.00000  3.46659
0.00000  0.00000  0.00000  3.46659

5.00000  0.00000  0.00000  3.20484
5.00000  0.00000  0.00000  3.20484
It seems that the data might need to be converted or processed differently to produce the odf.txt file required by the neper command. I would appreciate any guidance or explanation on how to achieve this conversion.

**Comment by rquey** [comment, marked as answer]
The values contained in `odf.txt` must correspond to the elemental values (of `fr.msh`).  So, you have to map your density field to that space / elements.

**Reply by ZXJ2024** [reply]
Thank you for your reply. Since I haven’t found relevant tutorials, could you explain in detail how a single element in `fr.msh` represents an orientation and how its orientation is determined or calculated based on the node coordinates?

**Reply by rquey** [reply]
```
neper -T -n 1 -domain "rodrigues(cubic)" -o fr
neper -M fr.tess -statelt x,y,z
```

The `fr.stelt` file will provide you with the element centers, which are orientations expressed as Rodrigues vectors.
